vcf2fhir/fhir_helper.py

```diff
@@ -264,162 +177,189 @@ def add_variant_obv(self, record, ref_seq, ratio_ad_dp):
         )
         observation_dv.component = []
 
-        observation_dv_component2 = observation.ObservationComponent()
-        observation_dv_component2.code = concept.CodeableConcept(
-            {
-                "coding": [
-                    {
-                        "system": "http://loinc.org",
-                        "code": "48013-7",
-                        "display": "Genomic reference sequence ID"
-                    }
-                ]
-            }
-        )
-        observation_dv_component2\
-            .valueCodeableConcept = concept.CodeableConcept(
-                {
-                    "coding": [
+        if record.is_sv:
+            dna_chg = get_dna_chg(record.INFO['SVTYPE'])
+            dna_change_type_component = observation.ObservationComponent()
+            dna_change_type_component.code = get_codeable_concept(
+                "http://loinc.org", "48019-4", "DNA Change Type"
+            )
+            dna_change_type_component\
+                .valueCodeableConcept = get_codeable_concept(
+                    "http://sequenceontology.org", dna_chg['CODE'],
+                    dna_chg['DISPLAY']
+                )
+
+            if(record.INFO['SVTYPE'] in list(SVs - {'INV', 'INS'}) and
+               hasattr(record.samples[0].data, 'CN')):
+                copy_number_component = observation.ObservationComponent()
+                copy_number_component.code = get_codeable_concept(
+                    "http://loinc.org", "82155-3",
+                    "Genomic Structural Variant copy Number"
+                )
+                copy_number_component\
+                    .valueQuantity = quantity.Quantity(
                         {
-                            "system": "http://www.ncbi.nlm.nih.gov/nuccore",
-                            "code": ref_seq
+                            "system": "http://unitsofmeasure.org",
+                            "code": '1',
+                            "value": record.samples[0]["CN"]
                         }
-                    ]
-                }
-            )
-        observation_dv.component.append(observation_dv_component2)
+                    )
 
-        if alleles['CODE'] != "" or alleles['ALLELE'] != "":
-            observation_dv_component3 = observation.ObservationComponent()
-            observation_dv_component3.code = concept.CodeableConcept(
-                {
-                    "coding": [
+            if hasattr(record.INFO, 'CIPOS') and hasattr(record.INFO, 'CIEND'):
+                inner_start = record.POS + record.INFO['CIPOS'][1]
+                inner_end = record.INFO['END'] + abs(record.INFO['CIEND'][0])
+                outer_start = record.POS - abs(record.INFO['CIPOS'][0])
+                outer_end = record.INFO['END'] + record.INFO['CIEND'][1]
+                outer_start_end_component = observation.ObservationComponent()
+                outer_start_end_component.code = get_codeable_concept(

```

Comment:
What are we doing here? It's hard to understand the logic here.

Reply:
Here we check, if there is a `CIPOS` and a `CIEND` field in `record.INFO` and calculate **inner and outer
start and end** based on it, else we ignore the `outer-start-end` component and populate the `inner-start-end` component based on only the `INFO.END` value. I have also added a comment just above this code.